app/jobs/generate_pdf_job.rb

```diff
@@ -80,4 +83,17 @@ def pdf_path(data_dir:, file_name:)
   def pdf_file_name(file_name:)
     file_name.sub(/(\.xml|\.pdf)?$/i, '').concat('.pdf').to_s
   end
+
+  def valid_pdf?(pdf_file_path:)
+    return false unless File.exist?(pdf_file_path)
+
+    pdf = PDF::Reader.new(pdf_file_path)
+    pdf.pages.any?
+  rescue PDF::Reader::MalformedPDFError, ArgumentError
+    false
+  rescue PDF::Reader::UnsupportedFeatureError
+    # If a feature is unsupported pdf-reader has still checked for the structure/EOF marker,
+    # which raises MalformedPDFError.
+    File.exist?(pdf_file_path)
```

Comment:
Can we just return `true` here because we checked that the file exists on line 88?

Reply:
We can and we should, thanks. I squashed that assignment too, seemed unnecessary.